### PR TITLE
Torch unet n2s

### PR DESCRIPTION
**Pull request by AhmetCanSolak**
This PR attempts to implement n2s on new torch unet model.

### CODE REVIEW

File: aydin/nn/models/torch/demo/unet/n2s_2D_generic.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    import napari
+
+    viewer = napari.Viewer()  # no prior setup needed
+    viewer.add_image(image, name='image')
+    viewer.add_image(noisy, name='noisy')
+    viewer.add_image(denoised, name='denoised')
+    napari.run()
+
+
+if __name__ == '__main__':
```
Comment (AhmetCanSolak) [inline aydin/nn/models/torch/demo/unet/n2s_2D_generic.py:85]: This is the demo script that I am testing the current implementation, maybe it is time for @li-li-github to slowly start reviewing and maybe can point out where I am causing J-leakage. Any comments/feedback is appreciated @li-li-github !

Comment (AhmetCanSolak) [reply aydin/nn/models/torch/demo/unet/n2s_2D_generic.py:85]: no more any problem there

File: aydin/nn/models/torch/test/test_torch_models.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
+# flake8: noqa
+import numpy
+import torch
+
+from aydin.io.datasets import lizard, add_noise, camera, normalise
+from aydin.nn.models.torch.torch_res_unet import ResidualUNetModel
+from aydin.nn.models.torch.torch_unet import UNetModel, n2t_train, n2s_train
+
+
+def test_supervised_2D():
```
Comment (li-li-github) [inline aydin/nn/models/torch/test/test_torch_models.py:10]: Is this a test without adding noise?

File: aydin/nn/models/torch/test/test_torch_models.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        # (64, 64, 64, 1),
+        nb_unet_levels=2,
+        spacetime_ndim=3,
+    )
+    result = model3d(input_array)
+    assert result.shape == input_array.shape
+    assert result.dtype == input_array.dtype
+
+
+# def test_various_masking_3D():
```
Comment (li-li-github) [inline aydin/nn/models/torch/test/test_torch_models.py:110]: Just a reminder that you can delete the comments if there's no plan to reactivate them.

File: aydin/nn/models/torch/torch_unet.py [hunk 4]
```diff
@@ -146,20 +139,109 @@ def _decoder_convolutions(self):
         return convolutions
 
 
-def n2t_unet_train_loop(
+def n2s_train(
```
Comment (li-li-github) [inline aydin/nn/models/torch/torch_unet.py:142]: Is this function particular to unet model or can it accept other models?

File: aydin/nn/models/utils/n2s_dataset.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
+import numpy
+import torch
+from torch.utils.data import Dataset
+
+
+class N2SDataset(Dataset):
```
Comment (li-li-github) [inline aydin/nn/models/utils/n2s_dataset.py:6]: I think it's a good idea to involve masking inside a Dataset object.

**Review by li-li-github — APPROVED** [review]
It looks good. I left a few comments & questions but mostly no need to change.
Good to merge.

File: aydin/nn/models/torch/torch_unet.py (hunk 4 above)
Comment (AhmetCanSolak) [reply aydin/nn/models/torch/torch_unet.py:142]: it can accept other models

File: aydin/nn/models/torch/test/test_torch_models.py (hunk 2 above)
Comment (AhmetCanSolak) [reply aydin/nn/models/torch/test/test_torch_models.py:10]: yes

File: aydin/nn/models/utils/n2s_dataset.py (hunk 5 above)
Comment (AhmetCanSolak) [reply aydin/nn/models/utils/n2s_dataset.py:6]: I will make separate dataset for different masking approaches and names will reflect accordingly.
